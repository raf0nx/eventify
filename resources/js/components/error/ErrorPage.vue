<template>
	<div class="display-1 error--text">Page not found!</div>
</template>

<script lang="ts">
	import { Vue, Component } from "vue-property-decorator";

	@Component
	export default class ErrorPage extends Vue {}
</script>

<style>
</style>