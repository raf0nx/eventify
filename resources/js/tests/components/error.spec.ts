import { createLocalVue, shallowMount, Wrapper } from "@vue/test-utils";

import ErrorPage from "@components/error/ErrorPage.vue";

const ERROR_PAGE_HEADER = "Page not found!";

describe("Auth.vue", () => {
    const localVue = createLocalVue();
    let wrapper: Wrapper<ErrorPage>;

    beforeEach(() => {
        wrapper = shallowMount(ErrorPage, { localVue });
    });

    afterEach(() => {
        wrapper.destroy();
    });

    it("Should match snapshot", () => {
        // Assert
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Should contain error page header", () => {
        const header = wrapper.find(".display-1");
        expect(header.text()).toEqual(ERROR_PAGE_HEADER);
    });
});
