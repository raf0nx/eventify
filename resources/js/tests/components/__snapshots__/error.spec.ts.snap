// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth.vue Should match snapshot 1`] = `"<div class=\\"display-1 error--text\\">Page not found!</div>"`;
